plugins/cassandra/cassandraembed.cpp

```diff
@@ -432,7 +439,7 @@ class CassandraStatementInfo : public CInterface
     {
         stop();
         futures.kill();
-        delete semaphore;
+        semaphore.clear();
```

Comment:
I'm not sure of this, that your def of extra work is the same as mine - it should optimize out the subsequent and duplicate ::Release(ptr) implicit in the destructor as your explicit .clear() assigns ptr = NULL but it still had to copy this pointer to temp before doing so, this is extra work that I don't believe the compiler will optimize away, though this is the point I'm not sure on.